modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/table/Table.tsx

```diff
@@ -292,37 +301,101 @@ const SaveFDSFieldsModalContent = ({
 								<ManagementToolbar.Item className="nav-item-expand">
 									<ClayInput.Group>
 										<ClayInput.GroupItem>
+											{!focused && (
+												<ClayInput.GroupInsetItem
+													before
+													tag="span"
+												>
+													<ClayIcon
+														className="inline-item inline-item-before"
+														focusable="false"
+														role="presentation"
+														symbol="search"
+													/>
+												</ClayInput.GroupInsetItem>
+											)}
+
 											<ClayInput
-												insetAfter
+												insetAfter={Boolean(focused)}
+												insetBefore={!focused}
 												onChange={(event) =>
 													onSearch(event.target.value)
 												}
+												onFocus={() => setFocused(true)}
 												placeholder={Liferay.Language.get(
 													'search'
 												)}
 												type="text"
 												value={query}
 											/>
 
-											<ClayInput.GroupInsetItem
-												after
-												tag="span"
-											>
-												<ClayButtonWithIcon
-													aria-label={Liferay.Language.get(
-														'search'
-													)}
-													displayType="unstyled"
-													symbol="search"
-												/>
-											</ClayInput.GroupInsetItem>
+											{focused && (
+												<ClayInput.GroupInsetItem
+													after
+													tag="span"
+												>
+													<ClayButtonWithIcon
+														aria-label={Liferay.Language.get(
+															'clear-search'
+														)}
+														borderless
+														displayType="secondary"
+														monospaced={false}
+														onClick={() => {
+															setQuery('');
+															onSearch('');
+															setFocused(false);
+														}}
+														size="sm"
+														symbol="times"
+														title={Liferay.Language.get(
+															'clear-search'
+														)}
+													/>
+												</ClayInput.GroupInsetItem>
+											)}
 										</ClayInput.GroupItem>
 									</ClayInput.Group>
 								</ManagementToolbar.Item>
 							</ManagementToolbar.ItemList>
 						</ManagementToolbar.Container>
 
-						<div className="bg-light fields pb-2 pt-2">
+						{getSelectedFieldsCount() > 0 && (
+							<ClayResultsBar>
+								<ClayResultsBar.Item expand>
+									<span className="component-text text-truncate-inline">
+										<span className="text-truncate">
+											{getSelectedFieldsCount() > 1
+												? `${getSelectedFieldsCount()} ${Liferay.Language.get(
+														'items-selected'
+												  )}`
+												: `${getSelectedFieldsCount()} ${Liferay.Language.get(
+														'item-selected'
```

Comment:
Here it's better to use `x-item-selected` and `x-items-selected` language keys, combined with `sub()` function, in order to prevent a fixed order of concatenation in languages where the number does not come first